internal/downloader/download.go

```diff
@@ -135,7 +168,31 @@ func (d *Downloder) download(desc, urlToGet, destination string, mode os.FileMod
 			fmt.Fprintln(os.Stderr, " done.")
 		}),
 	)
+	hasher := sha256.New()
+
+	_, err = io.Copy(io.MultiWriter(f, bar, hasher), resp.Body)
+	if err != nil {
+		return fmt.Errorf(
+			"Error while downloading text of %s into file %s: %v",
+			urlToGet, temporaryDestinationFile.Name(), err)
+	}
 
-	_, err = io.Copy(io.MultiWriter(f, bar), resp.Body)
-	return err
+	shaURLToGet := urlToGet + ".sha256"
+	shaExpected, err := d.getContentsOfURL(shaURLToGet)
+	if err != nil {
+		return fmt.Errorf("Error while trying to get contents of %s: %v", shaURLToGet, err)
+	}
+	shaExpected = strings.TrimRight(shaExpected, "\n")
+	shaActual := hex.EncodeToString(hasher.Sum(nil))
+	if shaExpected != shaActual {
+		return fmt.Errorf("Error downloading URL %s: Expected SHA [%s], got SHA [%s]",
```

Comment:
You will have to add these lines before the `return`
```golang
temporaryDestinationFile.Close()
os.Remove(temporaryDestinationFile.Name())
```

More about that below

Moreover, you should return a specific error type (see above), the line will look like
https://github.com/flavio/kuberlr/blob/cf1ff33c5318330e1086645ed8446b26d8ca21bc/internal/finder/kubectl_finder.go#L74

Reply:
Returning the specific error is done in 17fe1e5 - it's a de facto subclass of `error`, as it implements `Error() string`.